ee/vellum_ee/workflows/display/nodes/vellum/conditional_node.py

```diff
@@ -166,7 +169,11 @@ def serialize_rule(
             rule_group_id = str(
                 condition_ids[idx].rule_group_id if condition_ids else uuid4_from_hash(f"{condition_id}|rule_group")
             )
-            source_handle_id = str(source_handle_ids.get(idx) or uuid4_from_hash(f"{node_id}|handles|{idx}"))
+            source_handle_id = (
+                str(source_handle_ids.get(idx))
+                if source_handle_ids.get(idx)
+                else cast(JsonObject, ports_by_index[idx])["id"]
```

Comment:
[nit] what if, for some reason, `ports_by_index[idx]` throws a KeyError? feels like we should do the three stage check just to be safe where we default to the previous default. May be simpler with three variants to just helper method this instead of continuing the ternary